### DISCUSSION TITLE
Is it possible to use a different forecasthorizon without retraining

### DISCUSSION THREAD

**Discussion by ams015** [Nixtla/neuralforecast#809] [Q&A] [answered]
I am looking at models such as TFT, PatchTST, DeepAR, NBeats, etc. And it see that in fitting the model, h is somewhat we need

      nf = NeuralForecast(models=[TFT(h=20, hidden_size=int(row['config/hidden_size']),n_head =row['config/n_head'] 
                scaler_type=  row['config/scaler_type'] ,    learning_rate=row['config/learning_rate'], max_steps=row['config/max_steps'],  batch_size  =int(row['config/batch_size']   )       ,
                windows_batch_size=int(row['config/windows_batch_size']), loss=MAE(),random_seed =row['config/random_seed'],input_size=row['config/input_size']  ,    step_size=row['config/step_size']   ,
                valid_loss=MAE(),    val_check_steps=50, early_stop_patience_steps=10,enable_progress_bar=True
    
                )
      nf.fit(df=train_diff, val_size=20)
      Y_hat_df = nf.predict()

My question is, if I want to forecast with a different h, do I need to retrain the model with that new h? Or is there a way to forecast without retraining?

**Comment by jmoralez** [comment, marked as answer]
Hey @ams015, thanks for using neuralforecast. It's necessary to retrain because the model architecture is defined based on the horizon.

**Reply by tibeler** [reply]
why?in the recurssive way like lstm, we can obtain y_hat(t+1)based on y_hat(t).  so when the horizon = 1 i can make another prediction based on the first prediction,so it no longer have to retrain?
